docs/user/security/secure-saved-objects.asciidoc

```diff
@@ -0,0 +1,90 @@
+
+[[encryption-key-rotation]]
+==== Encryption key rotation
+
+Many policies and best practices stipulate that encryption keys should be periodically rotated to decrease the amount of content encrypted with one key and hence limit the potential damage in case the key is compromised. {kib} allows you to rotate encryption keys whenever there is a need. 
+
+As a rule of thumb, when you change an encryption key you still keep the old one for some time. While {kib} will only use a new encryption key to encrypt all new and updated data it still may need to know the old one to decrypt data that was encrypted using that old key. Moreover, it's perfectly possible to have multiple old keys used only for decryption. It's worth noting that {kib} doesn't re-encrypt existing Saved Objects with the new encryption key automatically, re-encryption only happens when you update existing object or use bulk key rotation API explained later on.
+
+Here is how your `kibana.yml` can look like if you want to use key rotation functionality:
+
+[source,yaml]
+--------------------------------------------------------------------------------
+xpack.encryptedSavedObjects:
+  encryptionKey: "min-32-byte-long-NEW-encryption-key" <1>
+  keyRotation:
+    decryptionOnlyKeys: ["min-32-byte-long-OLD#1-encryption-key", "min-32-byte-long-OLD#2-encryption-key"] <2>
+--------------------------------------------------------------------------------
+
+<1> Encryption key {kib} will use to encrypt all new or updated Saved Objects, known as a primary encryption key.
+<2> A list of encryption keys {kib} will try to use only to decrypt existing Saved Objects if decryption with the primary encryption key isn't possible, known as decryption-only or secondary encryption keys.
+
+[NOTE]
+============================================================================
+You may also leverage this functionality if, by mistake, you ended up with multiple {kib} instances connected to the same {es} cluster that use different encryption keys. In this case you may have a mix of Saved Objects encrypted with different keys and every {kib} instance can deal only with a specific subset of objects. To fix this you'll need to choose a single primary encryption key for `xpack.encryptedSavedObjects.encryptionKey`, move all other encryption keys to `xpack.encryptedSavedObjects.keyRotation.decryptionOnlyKeys` and sync this configuration across all {kib} instances.
+============================================================================
+
+At some point you may want to dispose old encryption keys completely, but before that you should make sure there are no Saved Objects left that {kib} encrypted with the encryption key you're going to dispose. You can use bulk key rotation API for that purpose. It will figure out which existing Saved Objects still require decryption-only keys and re-encrypt them with the primary key. It may take a while if you have lots of such Saved Objects:
+
```

Comment:
What do you think about extracting this out to the [Saved Objects API docs](https://github.com/elastic/kibana/blob/1e3f14395c5e720c0e3a21370387e80d27b8b653/docs/api/saved-objects.asciidoc), and then link to it from here?

I think we should also update this to match the format of the other REST API docs ([example](https://github.com/elastic/kibana/blob/0b675b89084b18faa1db1ca99ecd500a78af8f57/docs/api/saved-objects/export.asciidoc))

Reply:
Yeah, that's a great idea! Will do.